File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
// 다시 안풀어도 됨.
function solution(left, right) {
    var answer = 0;
    for(let i = left; i<=right; i++){
        let temp = 0;
        for(let j = 1; j<=i; j++){
            if(i % j === 0) temp++;
        }
        
        // console.log(temp)
        
        if(temp % 2 === 0) answer+= i;
        else answer -= i;
    }
    return answer;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    return sum([i if cal_measure(i) else -i for i in range(left, right+1)])
    
def cal_measure(n):
    measure_num = len([i for i in range(1, n + 1) if n % i == 0])
    return True if measure_num % 2 == 0 else False